hydrosdk/modelversion.py

```diff
@@ -424,17 +404,43 @@ def from_json(cluster, model_json):
         )
 
     @staticmethod
-    def delete_by_id(cluster, model_id):
+    def create(cluster, name: str, contract: ModelContract, metadata: Optional[dict] = None) -> 'ModelVersion':
+        """
+        Creates modelversion on the server
+        :param cluster: active cluster
+        :param name: name of model
+        :param contract:
+        :param metadata:
+        :raises Exception: If server returned not 200
+        :return: modelversion
+        """
+        model = {
+            "name": name,
+            "contract": ModelContract_to_contract_dict(contract),
+            "metadata": metadata
+        }
+
+        resp = cluster.request(method="POST", url="/api/v2/externalmodel", json=model)
+        if resp.ok:
+            resp_json = resp.json()
+            mv_obj = ModelVersion.from_json(cluster=cluster, model_version=resp_json)
+            return mv_obj
+        raise ModelVersion.BadRequest(
+            f"Failed to create external model. External model = {model}. {resp.status_code} {resp.text}")
+
+    @staticmethod
+    def delete_by_model_id(cluster: Cluster, model_id: int) -> dict:
         """
-        Deletes model by id
+        Deletes modelversion by model id
         :param cluster: active cluster
-        :param model_id: model id
+        :param model_id: model version id
         :return: if 200, json. Otherwise None
         """
-        res = cluster.request("DELETE", Model.BASE_URL + "/{}".format(model_id))
+        res = cluster.request("DELETE", ModelVersion.BASE_URL + "/{}".format(model_id))
         if res.ok:
             return res.json()
-        return None
+
+        raise ModelVersion.BadRequest(f"Failed to list delete by id modelversions. {res.status_code} {res.text}")
 
     @staticmethod
     def list_models(cluster) -> list:
```

Comment:
I suggest changing `def list_models(cluster) -> list:` to     `def list_model_versions(cluster: Cluster) -> List[ModelVersion]:`